### PR TITLE
Tag import setting

### PR DESCRIPTION
**Pull request by Wolfsblvt**
### Popup rework (another small thing)
- Add onClose/onClosing event handlers as optional pass-in options for popups, to grab stuff right when the popup closes

### Tag import popup improvements
- Save "remember" setting of tag import popup
- Add user option to change the tag import setting
- Improve tag import popup with adding drilled down bogus folders as auto-added tags
- Extract tag import popup to template
- Force-open popup no matter the setting on char dropdown button option

### CODE REVIEW

File: public/scripts/power-user.js [hunk 1]
```diff
@@ -1562,6 +1563,12 @@ function loadPowerUserSettings(settings, data) {
         power_user.tokenizer = tokenizers.GPT2;
     }
 
+    // Clean up old/legacy settings
+    if (power_user.import_card_tags !== undefined) {
```
Comment (Cohee1207) [inline public/scripts/power-user.js:1567]: I felt we should respect the choice of existing users.
For new users - let them see the popup.

Comment (Wolfsblvt) [reply public/scripts/power-user.js:1567]: ._____.

Existing users might have just been annoyed by the badly designed popup which was especially spammy for multi character import and clicked "Never" out of desparation. But would love to actually have a reasonable default setting while still not seeing the popup. If they knew about it.  
Before the choice was "Always blocking popup" or "no import".  
I can see several reasons why people who chose "no import" didn't actually want that option.

But well. We can still implement a mapping and just hide the new options for those people. I just think they'll never see it, which is a pity.

**Review by Cohee1207 — APPROVED** [review]
LGTM.
Fixed one minor thing and rewrote setting migration.

I guess those who already had tag import disabled shouldn't be nagged with it. It requires whole two clicks to never see it again.

**Comment by Wolfsblvt** [comment]
Still think it'll hurt more people who will miss the new features completely than it hurts people simply disabling it with two clicks again.

But you have more experience with the community here, so you might be right (:

**Comment by Cohee1207** [comment]
I guess there could be more certified card tag embed haters like me, who'd rather pretend it all never happened.

**Comment by Cohee1207** [comment]
It's also possible to add a line to patch notes to encourage to try it, like "try out new tag import popup, go to user settings and switch to 'ask', we think you're gonna like it!".

**Comment by Wolfsblvt** [comment]
Patch notes sounds like a good compromise. Works for me.

You actually had to manually select the tag option beforehand anyway. I checked, for my instance the settings value didn't even exist.

**Pull request merged**